subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/daemon/AbstractDaemonCompiler.java

```diff
@@ -72,6 +77,16 @@ public WorkResult execute(T spec) {
         }
     }
 
+    protected BaseForkOptions mergeForkOptions(BaseForkOptions left, BaseForkOptions right) {
+        BaseForkOptions merged = new BaseForkOptions();
+        merged.setMemoryInitialSize(mergeHeapSize(left.getMemoryInitialSize(), right.getMemoryInitialSize()));
+        merged.setMemoryMaximumSize(mergeHeapSize(left.getMemoryMaximumSize(), right.getMemoryMaximumSize()));
+        Set<String> mergedJvmArgs = normalized(left.getJvmArgs());
+        mergedJvmArgs.addAll(normalized(right.getJvmArgs()));
+        merged.setJvmArgs(Lists.newArrayList(mergedJvmArgs));
```

Comment:
Can we really merge the Jvm options? Wouldn't it possible that one part of the jvm options are for `javac` while the other part is for `java`?